pac4j-oidc/src/main/java/org/pac4j/oidc/logout/AjaxAwareOidcLogoutActionBuilder.java

```diff
@@ -0,0 +1,75 @@
+package org.pac4j.oidc.logout;
+
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.openid.connect.sdk.LogoutRequest;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.redirect.RedirectAction;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.Pac4jConstants;
+import org.pac4j.core.http.AjaxRequestResolver;
+import org.pac4j.core.http.DefaultAjaxRequestResolver;
+import org.pac4j.oidc.config.OidcConfiguration;
+import org.pac4j.oidc.profile.OidcProfile;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Logout action builder for OpenID Connect that takes care of ajax calls.
+ *
+ * @author Andrea Peruffo
+ * @since 2.2.0
+ */
+public class AjaxAwareOidcLogoutActionBuilder<U extends OidcProfile> extends OidcLogoutActionBuilder<U> {
+
+    private AjaxRequestResolver ajaxRequestResolver = new DefaultAjaxRequestResolver();
+
+    public AjaxAwareOidcLogoutActionBuilder() {}
+
+    public AjaxAwareOidcLogoutActionBuilder(OidcConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public RedirectAction getLogoutAction(final WebContext context, final U currentProfile, final String targetUrl) throws HttpAction {
+        init(context);
+
+        final String logoutUrl = configuration.getLogoutUrl();
+        if (CommonHelper.isNotBlank(logoutUrl)) {
+            try {
+                final URI endSessionEndpoint = new URI(logoutUrl);
+                final JWT idToken = currentProfile.getIdToken();
+
+                LogoutRequest logoutRequest;
+                if (CommonHelper.isNotBlank(targetUrl)) {
+                    logoutRequest = new LogoutRequest(endSessionEndpoint, idToken, new URI(targetUrl), null);
+                } else {
+                    logoutRequest = new LogoutRequest(endSessionEndpoint, idToken);
+                }
+
+                if (ajaxRequestResolver.isAjax(context)) {
```

Comment:
Why not put this change in the already existing `OidcLogoutActionBuilder`?

Reply:
ok, going to merge into it